apps/system/js/app_window.js

```diff
@@ -2126,14 +2126,22 @@
   };
 
   AppWindow.prototype._handle__cardviewbeforeshow = function aw_cvbeforeshow() {
-    this.debug('showing screenshot for cardsview.');
-    this._showScreenshotOverlay();
+    if (this.isActive()) {
+      // Hopefully update the screenshot in time for the cards view to show
+      this.debug('requesting, then showing screenshot for cardsview.');
+      this.getScreenshot(this._showScreenshotOverlay.bind(this));
+    } else {
+      this.debug('showing screenshot for cardsview.');
+      this._showScreenshotOverlay();
+    }
   };
 
   AppWindow.prototype._handle__cardviewshown = function aw_cvshown() {
+    this._showScreenshotOverlay();
```

Comment:
This guarantees the trigger (in case this.getScreenshot takes a long time to complete).